src/codegen/compile.ml

```diff
@@ -7166,11 +6937,10 @@ module MakeSerialization (Strm : Stream) = struct
         E.else_trap_with env "null rel_buf" ^^
         get_rel_buf ^^
         get_typtbl1 ^^
-        get_typtbl env ^^
         get_typtbl_end1 ^^
-        get_typtbl_end env ^^
         get_typtbl_size1 ^^
-        get_typtbl_size env ^^
+        load_candid_data env ^^
+        load_type_offsets env ^^
         get_idltyp1 ^^
```

Comment:
Too expensive?

Reply:
Indeed, I also need to optimize the the access functions (similar to the object pooling). This is probably the reason that idl subtype checks are that expensive. Thanks for noticing this.

Reply:
**Update**: I optimized the accesses to the global type data, such that the blobs are only loaded once at initialization. 

While measuring with https://github.com/dfinity/motoko/pull/4481, I have seen there is still some performance regression compared to classical persistence that I will need to analyze and if possible and reasonable, optimize.

Reply:
Please see above: There is still dynamic blob allocation in `idl_sub` that could be optimized for repeated calls in the same deserialization procedure...

Reply:
Thanks a lot for preparing https://github.com/dfinity/motoko/pull/4497. This will heal the regression.